src/test/java/project/service/GasStationServiceImplTest.java

```diff
@@ -82,28 +80,34 @@ void getGasPriceInfoTestWithThreeStations(String fuelType) {
         // GIVEN
         setupFunctionalityOfFindAllByFuelTypeWithThreeGasStations(fuelType);
 
-        // WHEN
-        GasPriceInfoDTO gasPriceInfo = gasStationService.getGasPriceInfo(fuelType);
-
-        // THEN
-        assertEquals(BigDecimal.valueOf(2.0).setScale(2, RoundingMode.HALF_UP), gasPriceInfo.getMedian());
-        assertEquals(BigDecimal.valueOf(1.0), gasPriceInfo.getMin());
-        assertEquals(BigDecimal.valueOf(3.0), gasPriceInfo.getMax());
+        // WHEN THEN
+        assertThat(gasStationService.getGasPriceInfo(fuelType))
+                .extracting("median")
+                .isEqualTo(BigDecimal.valueOf(2.0).setScale(2, RoundingMode.HALF_UP));
+        assertThat(gasStationService.getGasPriceInfo(fuelType))
+                .extracting("min")
+                .isEqualTo(BigDecimal.valueOf(1.0));
+        assertThat(gasStationService.getGasPriceInfo(fuelType))
+                .extracting("max")
+                .isEqualTo(BigDecimal.valueOf(3.0));
     }
 
     @ParameterizedTest
     @ValueSource(strings = {"e5", "e10", "diesel"})
-    void getGasPriceInfoTestWithTwoStations(String text) {
+    void getGasPriceInfoTestWithTwoStations(String fuelType) {
         // GIVEN
-        setupFunctionalityOfFindAllByFuelTypeWithTwoGasStations(text);
+        setupFunctionalityOfFindAllByFuelTypeWithTwoGasStations(fuelType);
 
-        // WHEN
-        GasPriceInfoDTO gasPriceInfo = gasStationService.getGasPriceInfo(text);
-
-        // THEN
-        assertEquals(BigDecimal.valueOf(1.5).setScale(2, RoundingMode.HALF_UP), gasPriceInfo.getMedian());
-        assertEquals(BigDecimal.valueOf(1.0), gasPriceInfo.getMin());
-        assertEquals(BigDecimal.valueOf(2.0), gasPriceInfo.getMax());
+        // WHEN THEN
```

Comment:
here three calls for the same - `gasStationService.getGasPriceInfo()`

Reply:
extracted and fixed